pkg/providers/tinkerbell/create.go

```diff
@@ -45,7 +45,15 @@ func (p *Provider) PreCAPIInstallOnBootstrap(ctx context.Context, cluster *types
 }
 
 func (p *Provider) PostBootstrapSetup(ctx context.Context, clusterConfig *v1alpha1.Cluster, cluster *types.Cluster) error {
-	return p.applyHardware(ctx, cluster)
+	if err := p.applyHardware(ctx, cluster); err != nil {
+		return err
+	}
+	hardwareSpec, err := p.generateHardwareSpec(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("generating hardware spec: %v", err)
+	}
+	p.hardwareSpec = hardwareSpec
```

Comment:
What's the motivation behind storing the hardwareSpec? Will this be used in the controller to get available hardware or something for workload clusters?

Reply:
We want to enable hardware management via GitOps and consequently upload the hardware YAML to the git repo. This PR is the first part of that initiative, and a subsequent PR will use this to upload the file to GitOps. Check the draft [here](https://github.com/aws/eks-anywhere/pull/4906) for details.

I've updated the description to include the motive.